src/App.tsx

```diff
@@ -1,61 +1,127 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [selectedUser, setSelectedUser] = useState(false);
+  const [enteredTitle, setEnteredTitle] = useState(false);
+
+  const newTodoId = Math.max(...todos.map(todo => todo.id)) + 1;
+
+  const addNewTodo = (id: number) => {
+    if (getUser(id) && title.trim()) {
+      todos.push({
+        id: newTodoId,
+        userId,
+        title,
+        completed: false,
+        user: getUser(userId),
+      });
+
+      setUserId(0);
+      setTitle('');
+    }
+
+    if (!userId) {
+      setUserId(0);
+      setSelectedUser(true);
+    }
+
+    if (!title.trim()) {
+      setEnteredTitle(true);
+    }
+  };
+
+  const onChangeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setEnteredTitle(false);
+  };
+
+  const handleUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setSelectedUser(false);
+  };
+
+  const handleForm = (event:React.ChangeEvent<HTMLFormElement>) => {
```

Comment:
```suggestion
  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setTitle(event.target.value);
    setEnteredTitle(false);
  };

  const handleUserSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
    setUserId(+event.target.value);
    setSelectedUser(false);
  };

  const handleFormSubmit = (event:React.ChangeEvent<HTMLFormElement>) => {
```
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364

Reply:
woow, it's so cool, I'm so grateful to you for this article...because as you noticed I have some mess up with the naming) But now I have clarity of what and how naming handle functions!)